File: src/MainMenuDemo/MenuButton.cs
using System;

namespace MainMenuDemo
{
    public class MenuButton
    {
        public string Label { get; set; }
        public string AssetPath { get; set; }

        public MenuButton(string label, string assetPath)
        {
            Label = label;
            AssetPath = assetPath;
        }

        public void Render()
        {
            Console.WriteLine($"Button: {Label} (Asset: {AssetPath})");
        }
    }
}

File: src/GameEngine/MenuButton.cpp
// MenuButton.cpp
#include "MenuButton.h"
#include <iostream>

MenuButton::MenuButton(const std::string& label, const std::string& assetPath)
    : label(label), assetPath(assetPath) {}

void MenuButton::Render() {
    std::cout << "Button: " << label << " (Asset: " << assetPath << ")" << std::endl;
}
